src/blockdev.rs

```diff
@@ -876,6 +877,12 @@ pub fn get_block_device_size(file: &File) -> Result<NonZeroU64> {
     }
 }
 
+/// Get the size of the GPT metadata at the start of the disk.
+pub fn get_gpt_size(file: &mut (impl Read + Seek)) -> Result<u64> {
```

Comment:
Unless I'm missing some case, I think you can directly take a `&GPT` to avoid scanning the disk twice everywhere.

Reply:
The callers that aren't in `blockdev.rs` (including the sole caller that isn't a test case) don't have a `&GPT`, and I think it'd be better if they don't have to know about `gptman`.